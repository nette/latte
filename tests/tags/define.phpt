<?php

/**
 * Test: {define ...}
 */

declare(strict_types=1);

use Tester\Assert;

require __DIR__ . '/../bootstrap.php';


$latte = new Latte\Engine;
$latte->setLoader(new Latte\Loaders\StringLoader);

$template = <<<'XX'
	{var $var = 10}

	{define test}
		This is definition #{$var}
	{/define}

	{include #test, var => 20}

	{define true}true{/define}
	{include true}
	XX;

Assert::matchFile(
	__DIR__ . '/expected/define.php',
	$latte->compile($template),
);
Assert::matchFile(
	__DIR__ . '/expected/define.html',
	$latte->renderToString($template),
);
