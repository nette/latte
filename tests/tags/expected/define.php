<?php
%A%
final class Template%a% extends Latte\Runtime\Template
{
	public const Blocks = [
		['test' => 'blockTest', 'true' => 'blockTrue'],
	];


	public function main(array $ʟ_args): void
	{
		extract($ʟ_args);
		unset($ʟ_args);

		echo '

';
		$this->renderBlock('test', ['var' => 20] + get_defined_vars(), 'html') /* line 7 */;
		echo '

';
		$this->renderBlock('true', get_defined_vars(), 'html') /* line 10 */;
	}


	public function prepare(): array
	{
		extract($this->params);

		$var = 10 /* line 1 */;
		return get_defined_vars();
	}


	/** {define test} on line 3 */
	public function blockTest(array $ʟ_args): void
	{
		extract($this->params);
		extract($ʟ_args);
		unset($ʟ_args);

		echo '	This is definition #';
		echo LR\Filters::escapeHtmlText($var) /* line 4 */;
		echo "\n";
	}


	/** {define true} on line 9 */
	public function blockTrue(array $ʟ_args): void
	{
		echo 'true';
	}
}
