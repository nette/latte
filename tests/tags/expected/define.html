

	This is definition #20


true
